pandas/core/arrays/datetimes.py

```diff
@@ -1903,32 +1904,36 @@ def maybe_convert_dtype(data, copy):
     ------
     TypeError : PeriodDType data is passed
     """
-    if is_float_dtype(data):
+    if not hasattr(data, "dtype"):
```

Comment:
is this part of the bugfix or additional cleanup?

Reply:
Cleanup for perf; my eye twitches whenever i see is_foo_dtype(values) that could be is_foo_dtype(values.dtype)